app/controllers/sessions_controller.rb

```diff
@@ -35,6 +35,7 @@ def create_from_json
   end
 
   def destroy_from_json
+    revoke_access_tokens! if doorkeeper_token
```

Comment:
Perhaps clarify what this method will do - we're revoking all tokens that this user has for the token's linked application, i.e. PFE or other client application.
```suggestion
    revoke_all_user_application_access_tokens! if doorkeeper_token
```

Reply:
You're testing for the token existence here and in the revoke method (via safe & operator), perhaps simplify the logic here and push the boolean existence test into the method call?

Personally i'd leave it and remove the safe operator as method calls aren't free so avoiding them is better!